### PR TITLE
add end story feature

### PR DESCRIPTION
**Pull request by Shiv-Patil**
I don't know if this will work... don't have much experience with discord.py and this is probably not the best way to do this.

### CODE REVIEW

File: cogs/fun_channel_checks.py [hunk 1]
```diff
@@ -43,7 +43,16 @@ async def on_message(self, message):
...
+            if message.content.lower() == "the end":
+                messages = await channel.history(limit=200)
+                fetch_message = messages.find(lambda m: m.author == client.user)
+                if fetch_message is None:
+                    await channel.send(f"{message.author.mention}, The last story was too large... Okay we are starting a new story, Let me start,\n\n{article}")
+                    return
+                story = " ".join(map(lambda m: m.content if (fetch_message.created_at < m.created_at) else '', messages))
+                await channel.send(story)
+                await channel.send(f"Okay we are starting a new story, Let me start,\n\n{article}")
+                return
```
Comment (classPythonAddike) [inline cogs/fun_channel_checks.py]: Over here you can do `return await channel.send(. . . . . )`, it'll look neater 👀 😃

File: cogs/fun_channel_checks.py [hunk 2]
```diff
@@ -43,7 +43,16 @@ async def on_message(self, message):
                         break
                     await message.delete()
                     return
-
+            if message.content.lower() == "the end":
+                messages = await channel.history(limit=200)
+                fetch_message = messages.find(lambda m: m.author == client.user)
```
Comment (classPythonAddike) [inline cogs/fun_channel_checks.py]: Also, this should be `lambda m: m.author == self.client.user` as you're doing this in a cog

File: cogs/fun_channel_checks.py (hunk 1 above)
Comment (Shiv-Patil) [reply cogs/fun_channel_checks.py]: ah :mhm:

File: cogs/fun_channel_checks.py (hunk 2 above)
Comment (Shiv-Patil) [reply cogs/fun_channel_checks.py]: o ok

File: cogs/fun_channel_checks.py [hunk 3]
```diff
@@ -45,14 +45,13 @@ async def on_message(self, message):
                     return
             if message.content.lower() == "the end":
                 messages = await channel.history(limit=200)
-                fetch_message = messages.find(lambda m: m.author == client.user)
+                fetch_message = messages.find(lambda m: m.author == self.client.user)
                 if fetch_message is None:
                     await channel.send(f"{message.author.mention}, The last story was too large... Okay we are starting a new story, Let me start,\n\n{article}")
                     return
```
Comment (classPythonAddike) [inline cogs/fun_channel_checks.py]: lol you can do the same on line 51 😂

Comment (Shiv-Patil) [reply cogs/fun_channel_checks.py]: who cares

Comment (classPythonAddike) [reply cogs/fun_channel_checks.py]: f.tag nou

**Comment by FirePlank** [comment]
I don't really think printing the whole story is required, also this is a very simple if check and then just start a new story so i don't think I will be merging this for the sake of that :( sorry

**Pull request closed without merging**